sidebars: Update unread pills to handle zoom better.

maxnuss opened the pull request:
This updates the unread pills in the left and right sidebar to look better at a wide range of zoom values. (It doesn't change their appearance at all.) It also refactors the CSS so that less code is duplicated between the multiple types of unread pill.

timabbott commented on static/styles/left-sidebar.scss:
```diff
@@ -172,12 +172,15 @@ li.active-sub-filter {
+    padding: 0 4px;
+    height: 15px;
+    line-height: 14px;
+    color: #fff;
```
I think we want to use the `hsl` color style here (as we do in the rest of the app) so that we don't need an entry in `night_mode.css` for this style (at least, we attempted to purge use of colors specified this way from the codebase).  

Speaking of which, we should probably have a lint rule for that; I opened https://github.com/zulip/zulip/issues/9689 to track that issue.

maxnuss commented:
Updated!

maxnuss commented:
Just updated this with a few minor improvements:
- Made the span height and line-height match so the text stays centered
- Made the height an even number of pixels (16px vs 15) so it scales better
- Adjusted the top position of #global_filters .count so it's centered in the selection highlight

timabbott commented:
This works great, merged, thanks @maxnuss!
